<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2023-08-11" />

<title>cTWAS_analysis_for_white_blood_counts</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">m6A_in_disease_genetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/m6A_in_disease_genetics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">cTWAS_analysis_for_white_blood_counts</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2023-08-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>m6A_in_disease_genetics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230331code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230331)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230331code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230331)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/projects/m6A_in_disease_genetics/code/ctwas/ctwas_config_b37.R
</td>
<td style="text-align:left;">
code/ctwas/ctwas_config_b37.R
</td>
</tr>
<tr>
<td style="text-align:left;">
~/projects/m6A_in_disease_genetics/code/ctwas/qiansheng/locus_plot.R
</td>
<td style="text-align:left;">
code/ctwas/qiansheng/locus_plot.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gum6Aindiseasegeneticstreed1d6b2a685645cd43947bf5b65765d32d0cd9a2atargetblankd1d6b2aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/m6A_in_disease_genetics/tree/d1d6b2a685645cd43947bf5b65765d32d0cd9a2a" target="_blank">d1d6b2a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gum6Aindiseasegeneticstreed1d6b2a685645cd43947bf5b65765d32d0cd9a2atargetblankd1d6b2aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/m6A_in_disease_genetics/tree/d1d6b2a685645cd43947bf5b65765d32d0cd9a2a" target="_blank">d1d6b2a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/figure/
    Ignored:    analysis/m6A_switch_to_disease_h2g.nb.html
    Ignored:    data/plots/

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/IBD_E_S_m6A.Rmd
    Untracked:  analysis/IBD_E_S_m6A_output.Rmd
    Untracked:  analysis/LDL_E_S_m6A.Rmd
    Untracked:  analysis/LDL_m6A_output.Rmd
    Untracked:  analysis/RA_m6A_output.Rmd
    Untracked:  analysis/WhiteBlood_WholeBlood_E_M.Rmd
    Untracked:  analysis/lymph_m6A_output.Rmd
    Untracked:  analysis/lymph_m6A_output_hg19.Rmd
    Untracked:  analysis/pre_weights_m6AQTL.txt
    Untracked:  analysis/rbc_E_S_m6A_output.Rmd
    Untracked:  analysis/rbc_m6A_output.Rmd
    Untracked:  analysis/wbc_E_S_m6A_output.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/all_m6a_sites_with_paired_cisNATs_summary.csv
    Untracked:  code/check_double_strand.ipynb
    Untracked:  code/check_double_strand_v2.ipynb
    Untracked:  code/ctwas/
    Untracked:  code/figure/
    Untracked:  code/learn_gviz.Rmd
    Untracked:  code/learn_gviz.html
    Untracked:  code/learn_gviz.nb.html
    Untracked:  code/m6AQTL_finemapping.Rmd
    Untracked:  code/summary_TWAS_coloc_m6A_2023.Rmd
    Untracked:  code/test_gviz.ipynb
    Untracked:  code/twas_genes_PP4_0.3_immune_traits_trackplots.pdf
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/ADCY7_gwas_input.tsv
    Untracked:  data/ADCY7_qtl_input.tsv
    Untracked:  data/Allergy_full_coloc.txt
    Untracked:  data/Asthma_full_coloc.txt
    Untracked:  data/CAD_full_coloc.txt
    Untracked:  data/Eosinophil_count_full_coloc.txt
    Untracked:  data/GSE125377_jointPeakReadCount.txt
    Untracked:  data/IBD_full_coloc.txt
    Untracked:  data/JointPeaks.bed
    Untracked:  data/Li2022_dsRNAs.xlsx
    Untracked:  data/Lupus_full_coloc.txt
    Untracked:  data/RA_full_coloc.txt
    Untracked:  data/TABLE1_hg19.txt
    Untracked:  data/TABLE1_hg19.txt.zip
    Untracked:  data/__MACOSX/
    Untracked:  data/coloc_blood_traits.csv
    Untracked:  data/crohns_disease_full_coloc.txt
    Untracked:  data/edit_sites_and_GE_neg_correlated.txt
    Untracked:  data/edit_sites_and_GE_pos_correlated.txt
    Untracked:  data/features
    Untracked:  data/human_EERs.csv
    Untracked:  data/human_EERs.txt
    Untracked:  data/lymph_full_coloc.txt
    Untracked:  data/m6A_TWAS_results.csv
    Untracked:  data/m6a_TWAS_genes.txt
    Untracked:  data/m6a_joint_calling_peaks.csv
    Untracked:  data/nat_sense_pairs.csv
    Untracked:  data/plt_full_coloc.txt
    Untracked:  data/rbc_full_coloc.txt
    Untracked:  data/rdw_full_coloc.txt
    Untracked:  data/reported_AS_targets_S1.txt
    Untracked:  data/reported_AS_wanowska.txt
    Untracked:  data/sig_coloc_results/
    Untracked:  data/test_locuscomparer.pdf
    Untracked:  data/ulcerative_colitis_full_coloc.txt
    Untracked:  data/wbc_full_coloc.txt
    Untracked:  output/.ipynb_checkpoints/
    Untracked:  output/all_m6a_sites_with_cisNATs.csv
    Untracked:  output/all_m6a_sites_with_paired_cisNATs_summary.csv
    Untracked:  output/all_m6a_sites_with_paired_cisNATs_summary_PP40.3.csv
    Untracked:  output/all_m6a_sites_with_paired_cisNATs_summary_PP40.5.csv
    Untracked:  output/all_m6a_sites_with_paired_cis_NATs.csv
    Untracked:  output/fine_mapped_m6AQTLs_TWAS_genes_highPP4.rds
    Untracked:  output/gene_summary.csv
    Untracked:  output/immune_related_m6A_targets.csv
    Untracked:  output/m6aQTL_dsRNAs_PPP2R3C_PRORP.pdf
    Untracked:  output/m6a_peaks_nearby_dsRNAs.csv
    Untracked:  output/m6a_sites_near_all_dsRNAs_twas.csv
    Untracked:  output/m6a_sites_near_dsRNAs_coloc.csv
    Untracked:  output/m6a_sites_near_dsRNAs_twas.csv
    Untracked:  output/m6a_sites_near_dsRNAs_twas_summary.csv
    Untracked:  output/m6a_sites_overlapping_NAT_twas.csv
    Untracked:  output/m6a_sites_overlapping_dsRNAs_coloc.csv
    Untracked:  output/m6a_sites_overlapping_dsRNAs_twas.csv
    Untracked:  output/m6a_sites_overlapping_dsRegions.csv
    Untracked:  output/m6a_sites_overlapping_dsRegions_coloc.csv
    Untracked:  output/negatively_correlated_genes.txt
    Untracked:  output/postively_correlated_genes.txt
    Untracked:  output/rs1806261_RABEP1-NUP88_focused_locusview.pdf
    Untracked:  output/rs1806261_RABEP1-NUP88_locusview.pdf
    Untracked:  output/rs3177647_MAPKAPK5-AS1-MAPKAPK5_locusview.pdf
    Untracked:  output/rs3204541_DDX55-EIF2B1_locusview.pdf
    Untracked:  output/rs7184802_ADCY7-BRD7_locusview.pdf
    Untracked:  output/rs7184802_ADCY7_locuscompare.pdf
    Untracked:  output/twas_genes_PP4_0.3_immune_traits_trackplots.pdf
    Untracked:  output/twas_genes_PP4_0.5_blood_traits_trackplots.pdf
    Untracked:  output/twas_m6a_sites_with_all_cisNATs.RDS
    Untracked:  output/twas_m6a_sites_with_cisNATs_range.RDS
    Untracked:  output/twas_m6a_sites_with_the_nearest_cisNAT.RDS
    Untracked:  twas_genes_PP4_0.3_immune_traits_trackplots.pdf

Unstaged changes:
    Modified:   analysis/m6A_switch_to_disease_h2g.Rmd
    Modified:   analysis/wbc_m6A_output.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/wbc_m6A_output_hg19.Rmd</code>)
and HTML (<code>docs/wbc_m6A_output_hg19.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/m6A_in_disease_genetics/blob/d1d6b2a685645cd43947bf5b65765d32d0cd9a2a/analysis/wbc_m6A_output_hg19.Rmd" target="_blank">d1d6b2a</a>
</td>
<td>
Jing Gu
</td>
<td>
2023-08-11
</td>
<td>
wflow_publish(c("analysis/wbc_m6A_output_hg19.Rmd",
"analysis/index.Rmd",
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/m6A_in_disease_genetics/blob/c94ee10dd04ad7af2c26821e45fa70fd1d133517/analysis/wbc_m6A_output_hg19.Rmd" target="_blank">c94ee10</a>
</td>
<td>
Jing Gu
</td>
<td>
2023-08-11
</td>
<td>
wflow_publish(c("analysis/wbc_m6A_output_hg19.Rmd",
"analysis/index.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
<pre class="r"><code># top 1 method
res &lt;- impute_expr_z(z_snp, weight = weight, ld_R_dir = ld_R_dir,
                         method = NULL, outputdir = outputdir, outname = outname.e,
                         harmonize_z = T, harmonize_wgt = T, scale_by_ld_variance=F,
                         strand_ambig_action_z = &quot;recover&quot;, 
                         recover_strand_ambig_wgt = T
# lasso/elastic-net method
res &lt;- impute_expr_z(z_snp, weight = weight, ld_R_dir = ld_R_dir,
                         method = NULL, outputdir = outputdir, outname = outname.e,
                         harmonize_z = T, harmonize_wgt = T, scale_by_ld_variance=F,
                         strand_ambig_action_z = &quot;none&quot;, 
                         recover_strand_ambig_wgt = F</code></pre>
<p>GWAS: UK Biobank GWAS summary statistics - European individuals</p>
<p>Weights: FUSION weights using top1, lasso, or elastic-net models were
converted into PredictDB format and were not needed to do scaling when
running ctwas.</p>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><strong>cTWAS analysis on m6A alone</strong></p>
<pre><code>[1] &quot;Check convergence for the top1 model:&quot;
[1] &quot;Table of group size:&quot;
    SNP    gene 
8713250     888 
                                SNP      gene
estimated_group_prior     2.481e-04 1.227e-02
estimated_group_prior_var 1.920e+01 2.631e+01
estimated_group_pve       1.184e-01 8.178e-04
attributable_group_pve    9.931e-01 6.858e-03
[1] &quot;Check convergence for the lasso model:&quot;
[1] &quot;Table of group size:&quot;
    SNP    gene 
8713250     912 
                                SNP      gene
estimated_group_prior     2.414e-04 1.016e-02
estimated_group_prior_var 1.898e+01 3.699e+01
estimated_group_pve       1.139e-01 9.778e-04
attributable_group_pve    9.915e-01 8.513e-03</code></pre>
<pre><code>$top1</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$lasso</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Joint analysis of expression, splicing and m6A</strong></p>
<pre><code>[1] &quot;Check convergence for the top1 model when jointly analyzing expression, splicing and m6A:&quot;
[1] &quot;Table of group size before/after matching with UKBB SNPs:&quot;
                          SNP      eQTL     sQTL   m6AQTL
prior_group_size    9.324e+06 2005.0000 2191.000 918.0000
group_size          8.713e+06 1928.0000 2123.000 888.0000
percent_of_overlaps 9.345e-01    0.9616    0.969   0.9673
                                SNP      eQTL      sQTL    m6AQTL
estimated_group_prior     2.406e-04 8.895e-03  0.012934 1.236e-02
estimated_group_prior_var 1.858e+01 1.683e+01 36.589120 2.554e+01
estimated_group_pve       1.112e-01 8.236e-04  0.002867 7.999e-04
attributable_group_pve    9.612e-01 7.120e-03  0.024783 6.916e-03</code></pre>
<pre><code>$top1</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ctwas-results-from-individual-analysis-on-m6a"
class="section level2">
<h2>cTWAS results from individual analysis on m6A</h2>
<p><strong>Lasso model</strong></p>
<div id="top-m6a-modification-pip" class="section level3">
<h3>Top m6A modification pip</h3>
<pre><code>  genename region_tag susie_pip       z
1 SLC9A3R1      17_42    0.9473  -7.630
2  ZKSCAN5       7_61    0.7976   7.112
3    ADCY7      16_27    0.7817   4.382
4    TRIT1       1_25    0.7516   5.554
5  THEMIS2       1_19    0.7034   6.243
6   BTN3A3       6_20    0.6855 -13.445
7  WAC-AS1      10_20    0.6102  11.178</code></pre>
<p><strong>Summing up PIPs for m6A peaks located in the same
gene</strong></p>
<p>Top m6A PIPs by genes</p>
<pre><code># A tibble: 7 × 2
  genename total_susie_pip
  &lt;chr&gt;              &lt;dbl&gt;
1 SLC9A3R1           0.947
2 ZKSCAN5            0.798
3 ADCY7              0.782
4 TRIT1              0.752
5 THEMIS2            0.703
6 BTN3A3             0.686
7 WAC-AS1            0.615</code></pre>
</div>
</div>
<div id="top-expressionsplicingm6a-units" class="section level2">
<h2>Top expression/splicing/m6A units</h2>
<p>For m6A or splicing QTLs, they are assigned to the nearest genes (m6A
needs to be confirmed with Kevin).</p>
<p><strong>Top SNPs or genes with PIP &gt; 0.6</strong></p>
<pre><code>$eQTL
     genename susie_pip group region_tag
1913  CSNK1G1    0.9587  eQTL      15_29
1916 RAPGEFL1    0.7362  eQTL      17_23
132    NDUFS2    0.6213  eQTL       1_81

$m6AQTL
     genename susie_pip  group region_tag
4938 SLC9A3R1    0.9539 m6AQTL      17_42
4922  ZKSCAN5    0.7863 m6AQTL       7_61
4067  THEMIS2    0.7816 m6AQTL       1_19
4312    TRAM2    0.7091 m6AQTL       6_39

$sQTL
     genename susie_pip group region_tag
4000   RNF181    1.0000  sQTL       2_54
4011    MYO1G    0.9932  sQTL       7_33
2385    GSK3B    0.7962  sQTL       3_74
4033   PDLIM1    0.7387  sQTL      10_61
3788   ZNF428    0.6957  sQTL      19_30</code></pre>
<div id="top-m6a-modification-pip-1" class="section level3">
<h3>Top m6A modification pip</h3>
<p>ZKSCAN5: RNA Polymerase II Cis-Regulatory Region Sequence-Specific
DNA Binding (<a href="GO:0000978" class="uri">GO:0000978</a>). THEMIS2
is involved in the biological process T Cell Receptor Signaling Pathway
(<a href="GO:0050852" class="uri">GO:0050852</a>). BANF: DNA binding
factor|Regulation Of Innate Immune Response (<a href="GO:0045088"
class="uri">GO:0045088</a>). TRIT1 has the molecular function of
Catalytic Activity, Acting On A tRNA (<a href="GO:0140101"
class="uri">GO:0140101</a>). TRIT1 is involved in the biological process
RNA Modification (<a href="GO:0009451" class="uri">GO:0009451</a>).
S1PR2 is involved in the biological process Regulation Of Cell
Population Proliferation (<a href="GO:0042127"
class="uri">GO:0042127</a>). WAC has the molecular function of RNA
Polymerase II Complex Binding (<a href="GO:0000993"
class="uri">GO:0000993</a>). CD320 is involved in the biological process
Regulation Of B Cell Proliferation (<a href="GO:0030888"
class="uri">GO:0030888</a>).</p>
<pre><code>   genename region_tag susie_pip      z
1  SLC9A3R1      17_42    0.9539 -7.630
2   ZKSCAN5       7_61    0.7863  7.158
3   THEMIS2       1_19    0.7816  6.277
4     TRAM2       6_39    0.7091  5.233
5     BANF1      11_36    0.5786  6.174
6     TRIT1       1_25    0.5278  5.298
7     S1PR2       19_9    0.5220  9.939
8   WAC-AS1      10_20    0.4945 11.169
9    SQSTM1      5_108    0.4934 -4.857
10    CD320       19_8    0.3627 -4.062</code></pre>
<p><strong>Summing up PIPs for m6A peaks located in the same
gene</strong></p>
<p>Top 10 m6A PIPs by genes</p>
<pre><code># A tibble: 800 × 2
   genename total_susie_pip
   &lt;chr&gt;              &lt;dbl&gt;
 1 SLC9A3R1           0.954
 2 ZKSCAN5            0.786
 3 THEMIS2            0.782
 4 TRAM2              0.709
 5 BANF1              0.579
 6 TRIT1              0.528
 7 S1PR2              0.522
 8 WAC-AS1            0.513
 9 SQSTM1             0.493
10 CD320              0.379
# ℹ 790 more rows</code></pre>
</div>
<div id="top-splicing-pips" class="section level3">
<h3>Top splicing PIPs</h3>
<p>Some loci contain variants in the same credible set but having
opposite z scores. For instance, the predicted splicing levels of two
introns of CNN2 based on the same variant (position=1038445) have
opposite associations with traits. Is this variant more likely to affect
traits by altering the splicing levels of both transcripts, rather than
one of them since they have equal PIP?</p>
<pre><code>                    peak_id   genename       pos region_tag susie_pip       z
1    chr2:85823772-85824227     RNF181  85818886       2_54    1.0000   5.009
2    chr7:45009474-45009639      MYO1G  45009341       7_33    0.9932 -11.719
3  chr3:119582452-119624602      GSK3B 119542297       3_74    0.7962   5.622
4   chr10:97007123-97023621     PDLIM1  97023552      10_61    0.7387  -7.331
5   chr19:44112259-44118381     ZNF428  44146930      19_30    0.6957  -4.929
6  chr5:122111457-122130961       SNX2 122088686       5_74    0.5628  -6.687
7   chr11:67120548-67124214 AP003419.1  67185596      11_37    0.5416  -4.432
8    chr6:29693820-29694660      HLA-F  29688501       6_23    0.5000 -16.046
9    chr6:29694781-29695734      HLA-F  29688501       6_23    0.5000 -16.046
10  chr11:47761655-47765505      FNBP4  47863119      11_29    0.4878  10.101
11  chr19:13885521-13886291   C19orf53  13942221      19_11    0.4797   6.500
12  chr19:13886427-13888866   C19orf53  13942221      19_11    0.4797   6.500
13   chr7:72986365-72987174       TBL2  72989390       7_47    0.4741   6.870
14 chr1:224544695-224548197      CNIH4 224630695      1_116    0.4698   8.830
15    chr19:1036561-1037624       CNN2   1038445       19_2    0.4643   6.170
16    chr19:1036999-1037624       CNN2   1038445       19_2    0.4643  -6.170
17  chr17:47288203-47295101       ABI3  47287067      17_28    0.4626  -4.041
18  chr19:49458856-49459455        BAX  49459104      19_34    0.4605  -4.118
19     chr7:5569315-5570155       ACTB   5556807        7_7    0.4288  -4.696
20  chr16:67690548-67690704    CARMIL2  67780829      16_36    0.3912  -3.955</code></pre>
<p><strong>Summing up PIPs for spliced introns located in the same
gene</strong></p>
<p>Top 10 splicing PIPs by genes</p>
<pre><code># A tibble: 10 × 2
   genename total_susie_pip
   &lt;chr&gt;              &lt;dbl&gt;
 1 HLA-F              1.01 
 2 RNF181             1.00 
 3 MYO1G              0.993
 4 C19orf53           0.959
 5 CNN2               0.929
 6 CD46               0.904
 7 HNRNPK             0.811
 8 GSK3B              0.796
 9 ZNF428             0.748
10 PDLIM1             0.739</code></pre>
</div>
<div id="top-genes-by-combined-pip" class="section level3">
<h3>Top genes by combined PIP</h3>
<pre><code>     genename combined_pip expression_pip splicing_pip   m6A_pip region_tag
1456    HLA-F        1.006      6.563e-06       1.0057 1.512e-05       6_23
2437   RNF181        1.000      0.000e+00       1.0000 0.000e+00       2_54
1891    MYO1G        0.993      0.000e+00       0.9932 0.000e+00       7_33
349  C19orf53        0.982      0.000e+00       0.9594 2.284e-02      19_11
605   CSNK1G1        0.969      9.587e-01       0.0000 1.015e-02      15_29
2656 SLC9A3R1        0.954      0.000e+00       0.0000 9.539e-01      17_42
1468   HNRNPK        0.950      0.000e+00       0.8109 1.388e-01       9_41
548      CNN2        0.929      0.000e+00       0.9287 0.000e+00       19_2
454      CD46        0.904      0.000e+00       0.9044 0.000e+00      1_107
1392    GSK3B        0.796      0.000e+00       0.7962 0.000e+00       3_74
3218  ZKSCAN5        0.786      0.000e+00       0.0000 7.863e-01       7_61
2868  THEMIS2        0.782      0.000e+00       0.0000 7.816e-01       1_19
2986    TRAM2        0.755      4.565e-02       0.0000 7.091e-01       6_39
3252   ZNF428        0.748      0.000e+00       0.7475 0.000e+00      19_30
2114   PDLIM1        0.739      0.000e+00       0.7387 0.000e+00      10_61
2360 RAPGEFL1        0.736      7.362e-01       0.0000 0.000e+00      17_23
285    BCL2A1        0.706      0.000e+00       0.7064 0.000e+00      15_37
547     CNIH4        0.690      0.000e+00       0.6904 0.000e+00      1_116
277       BAX        0.653      0.000e+00       0.6532 0.000e+00      19_34
1944   NDUFS2        0.621      6.213e-01       0.0000 0.000e+00       1_81</code></pre>
<!-- ## GO enrichment analysis for genes with PIP>0.5 -->
<!-- ```{r echo=FALSE, message=FALSE, eval=FALSE} -->
<!-- #GO enrichment analysis -->
<!-- library(enrichR) -->
<!-- dbs <- c("GO_Biological_Process_2021", "GO_Cellular_Component_2021", "GO_Molecular_Function_2021") -->
<!-- genes <- df_gene$gene_id[df_gene$combined_pip>0.8] -->
<!-- ``` -->
<!-- ```{r echo=F, message=FALSE, eval=FALSE} -->
<!-- if (length(genes)>0){ -->
<!--   GO_enrichment <- enrichr(genes, dbs) -->
<!--   for (db in dbs){ -->
<!--     print(db) -->
<!--     df <- GO_enrichment[[db]] -->
<!--     print(plotEnrich(GO_enrichment[[db]])) -->
<!--     df <- df[df$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")] -->
<!--     print(df) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ## DisGeNET enrichment analysis for genes with PIP>0.5 -->
<!-- ```{r echo=F, message=FALSE, eval=FALSE} -->
<!-- if (length(genes)>0){ -->
<!--   #library(disgenet2r) -->
<!--   disgenet_api_key <- get_disgenet_api_key( -->
<!--                     email = "wesleycrouse@gmail.com",  -->
<!--                     password = "uchicago1" ) -->
<!--   Sys.setenv(DISGENET_API_KEY= disgenet_api_key) -->
<!--   res_enrich <-disease_enrichment(entities=genes, vocabulary = "HGNC", -->
<!--                                database = "CURATED" ) -->
<!--   df <- res_enrich@qresult[1:10, c("Description", "FDR", "Ratio",  "BgRatio")] -->
<!--   print(df) -->
<!-- } -->
<!-- ``` -->
<!-- ## WebGestalt enrichment analysis for genes with PIP>0.5 -->
<!-- ```{r echo=F, message=FALSE, eval=FALSE} -->
<!-- if (length(genes)>0){ -->
<!--   #WebGestalt enrichment -->
<!--   library(WebGestaltR) -->
<!--   background <- ctwas_gene_res$genename -->
<!--   #listGeneSet() -->
<!--   databases <- c("pathway_KEGG", "disease_GLAD4U", "disease_OMIM") -->
<!--   enrichResult <- WebGestaltR(enrichMethod="ORA", organism="hsapiens", -->
<!--                               interestGene=genes, referenceGene=background, -->
<!--                               enrichDatabase=databases, interestGeneType="genesymbol", -->
<!--                               referenceGeneType="genesymbol", isOutput=F) -->
<!--   print(enrichResult[,c("description", "size", "overlap", "FDR", "database", "userId")]) -->
<!-- } -->
<!-- ``` -->
<pre><code>Loading required package: grid</code></pre>
<pre><code>Warning: replacing previous import &#39;utils::download.file&#39; by
&#39;restfulr::download.file&#39; when loading &#39;rtracklayer&#39;</code></pre>
</div>
</div>
<div id="locus-plots-for-specific-examples" class="section level2">
<h2>Locus plots for specific examples</h2>
<pre><code>     genename combined_pip expression_pip splicing_pip m6A_pip region_tag
1944   NDUFS2        0.621         0.6213            0       0       1_81</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>    genename combined_pip expression_pip splicing_pip m6A_pip region_tag
605  CSNK1G1        0.969         0.9587            0 0.01015      15_29</code></pre>
<pre><code>Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size
1.1 GiB</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>     genename combined_pip expression_pip splicing_pip m6A_pip region_tag
2868  THEMIS2        0.782              0            0  0.7816       1_19</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>     genename combined_pip expression_pip splicing_pip m6A_pip region_tag
3218  ZKSCAN5        0.786              0            0  0.7863       7_61</code></pre>
<p><img src="figure/wbc_m6A_output_hg19.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Gviz_1.40.1          cowplot_1.1.1        ggplot2_3.4.2       
 [4] GenomicRanges_1.48.0 GenomeInfoDb_1.32.2  IRanges_2.30.1      
 [7] S4Vectors_0.34.0     BiocGenerics_0.42.0  ctwas_0.1.38        
[10] dplyr_1.1.2          workflowr_1.7.0     

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0            deldir_1.0-6               
  [3] rjson_0.2.21                rprojroot_2.0.3            
  [5] biovizBase_1.44.0           htmlTable_2.4.0            
  [7] XVector_0.36.0              base64enc_0.1-3            
  [9] fs_1.6.3                    dichromat_2.0-0.1          
 [11] rstudioapi_0.15.0           farver_2.1.1               
 [13] bit64_4.0.5                 AnnotationDbi_1.58.0       
 [15] fansi_1.0.4                 xml2_1.3.3                 
 [17] codetools_0.2-18            logging_0.10-108           
 [19] cachem_1.0.8                knitr_1.39                 
 [21] Formula_1.2-4               jsonlite_1.8.7             
 [23] Rsamtools_2.12.0            cluster_2.1.3              
 [25] dbplyr_2.3.3                png_0.1-7                  
 [27] compiler_4.2.0              httr_1.4.6                 
 [29] backports_1.4.1             lazyeval_0.2.2             
 [31] Matrix_1.6-0                fastmap_1.1.1              
 [33] cli_3.6.1                   later_1.3.0                
 [35] htmltools_0.5.2             prettyunits_1.1.1          
 [37] tools_4.2.0                 gtable_0.3.3               
 [39] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [41] rappdirs_0.3.3              Rcpp_1.0.11                
 [43] Biobase_2.56.0              jquerylib_0.1.4            
 [45] vctrs_0.6.3                 Biostrings_2.64.0          
 [47] rtracklayer_1.56.0          iterators_1.0.14           
 [49] xfun_0.30                   stringr_1.5.0              
 [51] ps_1.7.0                    lifecycle_1.0.3            
 [53] ensembldb_2.20.2            restfulr_0.0.14            
 [55] XML_3.99-0.14               getPass_0.2-2              
 [57] zlibbioc_1.42.0             scales_1.2.1               
 [59] BSgenome_1.64.0             VariantAnnotation_1.42.1   
 [61] ProtGenerics_1.28.0         hms_1.1.3                  
 [63] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [65] parallel_4.2.0              SummarizedExperiment_1.26.1
 [67] AnnotationFilter_1.20.0     RColorBrewer_1.1-3         
 [69] yaml_2.3.5                  curl_5.0.1                 
 [71] memoise_2.0.1               gridExtra_2.3              
 [73] sass_0.4.1                  biomaRt_2.52.0             
 [75] rpart_4.1.16                latticeExtra_0.6-30        
 [77] stringi_1.7.12              RSQLite_2.3.1              
 [79] highr_0.9                   BiocIO_1.6.0               
 [81] foreach_1.5.2               checkmate_2.1.0            
 [83] GenomicFeatures_1.48.4      filelock_1.0.2             
 [85] BiocParallel_1.30.3         rlang_1.1.1                
 [87] pkgconfig_2.0.3             matrixStats_0.62.0         
 [89] bitops_1.0-7                evaluate_0.15              
 [91] lattice_0.20-45             htmlwidgets_1.5.4          
 [93] GenomicAlignments_1.32.0    labeling_0.4.2             
 [95] bit_4.0.5                   processx_3.8.0             
 [97] tidyselect_1.2.0            magrittr_2.0.3             
 [99] R6_2.5.1                    generics_0.1.3             
[101] Hmisc_5.1-0                 DelayedArray_0.22.0        
[103] DBI_1.1.3                   pgenlibr_0.3.6             
[105] pillar_1.9.0                whisker_0.4                
[107] foreign_0.8-82              withr_2.5.0                
[109] KEGGREST_1.36.2             RCurl_1.98-1.7             
[111] nnet_7.3-17                 tibble_3.2.1               
[113] crayon_1.5.2                interp_1.1-4               
[115] utf8_1.2.3                  BiocFileCache_2.4.0        
[117] rmarkdown_2.14              jpeg_0.1-10                
[119] progress_1.2.2              data.table_1.14.8          
[121] blob_1.2.4                  callr_3.7.3                
[123] git2r_0.30.1                digest_0.6.33              
[125] httpuv_1.6.5                munsell_0.5.0              
[127] bslib_0.3.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
