<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2023-08-10" />

<title>cTWAS_analysis_for_white_blood_counts</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">m6A_in_disease_genetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/m6A_in_disease_genetics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">cTWAS_analysis_for_white_blood_counts</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2023-08-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>m6A_in_disease_genetics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230331code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230331)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230331code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230331)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/projects/m6A_in_disease_genetics/code/ctwas/ctwas_config_b37.R
</td>
<td style="text-align:left;">
code/ctwas/ctwas_config_b37.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gum6Aindiseasegeneticstreef68611f23cf6124e16156d8f9616a5833df5c40atargetblankf68611fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/m6A_in_disease_genetics/tree/f68611f23cf6124e16156d8f9616a5833df5c40a" target="_blank">f68611f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gum6Aindiseasegeneticstreef68611f23cf6124e16156d8f9616a5833df5c40atargetblankf68611fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/m6A_in_disease_genetics/tree/f68611f23cf6124e16156d8f9616a5833df5c40a" target="_blank">f68611f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/m6A_switch_to_disease_h2g.nb.html
    Ignored:    data/plots/

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/IBD_E_S_m6A.Rmd
    Untracked:  analysis/IBD_E_S_m6A_output.Rmd
    Untracked:  analysis/LDL_E_S_m6A.Rmd
    Untracked:  analysis/LDL_m6A_output.Rmd
    Untracked:  analysis/RA_m6A_output.Rmd
    Untracked:  analysis/WhiteBlood_WholeBlood_E_M.Rmd
    Untracked:  analysis/learn_ctwas.Rmd
    Untracked:  analysis/lymph_m6A_output.Rmd
    Untracked:  analysis/pre_weights_m6AQTL.txt
    Untracked:  analysis/rbc_E_S_m6A_output.Rmd
    Untracked:  analysis/rbc_m6A_output.Rmd
    Untracked:  analysis/wbc_E_S_m6A_output.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/all_m6a_sites_with_paired_cisNATs_summary.csv
    Untracked:  code/check_double_strand.ipynb
    Untracked:  code/check_double_strand_v2.ipynb
    Untracked:  code/ctwas/
    Untracked:  code/figure/
    Untracked:  code/learn_gviz.Rmd
    Untracked:  code/learn_gviz.html
    Untracked:  code/learn_gviz.nb.html
    Untracked:  code/m6AQTL_finemapping.Rmd
    Untracked:  code/summary_TWAS_coloc_m6A_2023.Rmd
    Untracked:  code/test_gviz.ipynb
    Untracked:  code/twas_genes_PP4_0.3_immune_traits_trackplots.pdf
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/ADCY7_gwas_input.tsv
    Untracked:  data/ADCY7_qtl_input.tsv
    Untracked:  data/Allergy_full_coloc.txt
    Untracked:  data/Asthma_full_coloc.txt
    Untracked:  data/CAD_full_coloc.txt
    Untracked:  data/Eosinophil_count_full_coloc.txt
    Untracked:  data/GSE125377_jointPeakReadCount.txt
    Untracked:  data/IBD_full_coloc.txt
    Untracked:  data/JointPeaks.bed
    Untracked:  data/Li2022_dsRNAs.xlsx
    Untracked:  data/Lupus_full_coloc.txt
    Untracked:  data/RA_full_coloc.txt
    Untracked:  data/TABLE1_hg19.txt
    Untracked:  data/TABLE1_hg19.txt.zip
    Untracked:  data/__MACOSX/
    Untracked:  data/coloc_blood_traits.csv
    Untracked:  data/crohns_disease_full_coloc.txt
    Untracked:  data/edit_sites_and_GE_neg_correlated.txt
    Untracked:  data/edit_sites_and_GE_pos_correlated.txt
    Untracked:  data/features
    Untracked:  data/human_EERs.csv
    Untracked:  data/human_EERs.txt
    Untracked:  data/lymph_full_coloc.txt
    Untracked:  data/m6A_TWAS_results.csv
    Untracked:  data/m6a_TWAS_genes.txt
    Untracked:  data/m6a_joint_calling_peaks.csv
    Untracked:  data/nat_sense_pairs.csv
    Untracked:  data/plt_full_coloc.txt
    Untracked:  data/rbc_full_coloc.txt
    Untracked:  data/rdw_full_coloc.txt
    Untracked:  data/reported_AS_targets_S1.txt
    Untracked:  data/reported_AS_wanowska.txt
    Untracked:  data/sig_coloc_results/
    Untracked:  data/test_locuscomparer.pdf
    Untracked:  data/ulcerative_colitis_full_coloc.txt
    Untracked:  data/wbc_full_coloc.txt
    Untracked:  output/.ipynb_checkpoints/
    Untracked:  output/all_m6a_sites_with_cisNATs.csv
    Untracked:  output/all_m6a_sites_with_paired_cisNATs_summary.csv
    Untracked:  output/all_m6a_sites_with_paired_cisNATs_summary_PP40.3.csv
    Untracked:  output/all_m6a_sites_with_paired_cisNATs_summary_PP40.5.csv
    Untracked:  output/all_m6a_sites_with_paired_cis_NATs.csv
    Untracked:  output/fine_mapped_m6AQTLs_TWAS_genes_highPP4.rds
    Untracked:  output/gene_summary.csv
    Untracked:  output/immune_related_m6A_targets.csv
    Untracked:  output/m6aQTL_dsRNAs_PPP2R3C_PRORP.pdf
    Untracked:  output/m6a_peaks_nearby_dsRNAs.csv
    Untracked:  output/m6a_sites_near_all_dsRNAs_twas.csv
    Untracked:  output/m6a_sites_near_dsRNAs_coloc.csv
    Untracked:  output/m6a_sites_near_dsRNAs_twas.csv
    Untracked:  output/m6a_sites_near_dsRNAs_twas_summary.csv
    Untracked:  output/m6a_sites_overlapping_NAT_twas.csv
    Untracked:  output/m6a_sites_overlapping_dsRNAs_coloc.csv
    Untracked:  output/m6a_sites_overlapping_dsRNAs_twas.csv
    Untracked:  output/m6a_sites_overlapping_dsRegions.csv
    Untracked:  output/m6a_sites_overlapping_dsRegions_coloc.csv
    Untracked:  output/negatively_correlated_genes.txt
    Untracked:  output/postively_correlated_genes.txt
    Untracked:  output/rs1806261_RABEP1-NUP88_focused_locusview.pdf
    Untracked:  output/rs1806261_RABEP1-NUP88_locusview.pdf
    Untracked:  output/rs3177647_MAPKAPK5-AS1-MAPKAPK5_locusview.pdf
    Untracked:  output/rs3204541_DDX55-EIF2B1_locusview.pdf
    Untracked:  output/rs7184802_ADCY7-BRD7_locusview.pdf
    Untracked:  output/rs7184802_ADCY7_locuscompare.pdf
    Untracked:  output/twas_genes_PP4_0.3_immune_traits_trackplots.pdf
    Untracked:  output/twas_genes_PP4_0.5_blood_traits_trackplots.pdf
    Untracked:  output/twas_m6a_sites_with_all_cisNATs.RDS
    Untracked:  output/twas_m6a_sites_with_cisNATs_range.RDS
    Untracked:  output/twas_m6a_sites_with_the_nearest_cisNAT.RDS
    Untracked:  twas_genes_PP4_0.3_immune_traits_trackplots.pdf

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/m6A_switch_to_disease_h2g.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/wbc_m6A_output.Rmd</code>) and
HTML (<code>docs/wbc_m6A_output.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/m6A_in_disease_genetics/blob/f68611f23cf6124e16156d8f9616a5833df5c40a/analysis/wbc_m6A_output.Rmd" target="_blank">f68611f</a>
</td>
<td>
Jing Gu
</td>
<td>
2023-08-10
</td>
<td>
wflow_publish("~/projects/m6A_in_disease_genetics/analysis/wbc_m6A_output.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/m6A_in_disease_genetics/170a3106ef5ffa14fc1bb7d602feaa5da8b63e30/docs/wbc_m6A_output.html" target="_blank">170a310</a>
</td>
<td>
Jing Gu
</td>
<td>
2023-08-10
</td>
<td>
gene-association with ctwas
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-ctwas-results" class="section level3">
<h3>Load ctwas results</h3>
<pre class="r"><code># top 1 method
res &lt;- impute_expr_z(z_snp, weight = weight, ld_R_dir = ld_R_dir,
                         method = NULL, outputdir = outputdir, outname = outname.e,
                         harmonize_z = T, harmonize_wgt = T, scale_by_ld_variance=F,
                         strand_ambig_action_z = &quot;recover&quot;, 
                         recover_strand_ambig_wgt = T
# lasso/elastic-net method
res &lt;- impute_expr_z(z_snp, weight = weight, ld_R_dir = ld_R_dir,
                         method = NULL, outputdir = outputdir, outname = outname.e,
                         harmonize_z = F, harmonize_wgt = F, scale_by_ld_variance=T,
                         strand_ambig_action_z = &quot;recover&quot;, 
                         recover_strand_ambig_wgt = T</code></pre>
<p>GWAS: UK Biobank GWAS summary statistics - European individuals</p>
<p>Weights: FUSION weights using top1, lasso, or elastic-net models
converted into PredictDB format</p>
<p>Imputation: For the lasso and elastic-net models, I made mistakes in
not harmonizing QTL SNPs and GWAS/LD SNPs, and in performing scaling for
FUSION weights.</p>
</div>
<div id="check-convergence-of-parameters" class="section level3">
<h3>Check convergence of parameters</h3>
<p><strong>cTWAS analysis on m6A alone</strong></p>
<pre><code>[1] &quot;Check convergence for the top1 model:&quot;
[1] &quot;Table of group size:&quot;
    SNP    gene 
8713250     888 
                                SNP      gene
estimated_group_prior     2.481e-04 1.227e-02
estimated_group_prior_var 1.920e+01 2.631e+01
estimated_group_pve       1.184e-01 8.178e-04
attributable_group_pve    9.931e-01 6.858e-03
[1] &quot;Check convergence for the enet model:&quot;
[1] &quot;Table of group size:&quot;
    SNP    gene 
8713250     908 
                                SNP     gene
estimated_group_prior     2.375e-04  0.01375
estimated_group_prior_var 1.861e+01 31.17344
estimated_group_pve       1.099e-01  0.00111
attributable_group_pve    9.900e-01  0.01001
[1] &quot;Check convergence for the lasso model:&quot;
[1] &quot;Table of group size:&quot;
    SNP    gene 
8713250     912 
                                SNP      gene
estimated_group_prior      0.000241 1.180e-02
estimated_group_prior_var 19.018872 3.113e+01
estimated_group_pve        0.113956 9.563e-04
attributable_group_pve     0.991678 8.322e-03</code></pre>
<pre><code>$top1</code></pre>
<p><img src="figure/wbc_m6A_output.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$enet</code></pre>
<p><img src="figure/wbc_m6A_output.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$lasso</code></pre>
<p><img src="figure/wbc_m6A_output.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Joint analysis of expression, splicing and m6A</strong></p>
<pre><code>[1] &quot;Check convergence for the top1 model when jointly analyzing expression, splicing and m6A:&quot;
[1] &quot;Table of group size before/after matching with UKBB SNPs:&quot;
                          SNP      eQTL     sQTL   m6AQTL
prior_group_size    9.324e+06 2005.0000 2191.000 918.0000
group_size          8.713e+06 1928.0000 2123.000 888.0000
percent_of_overlaps 9.345e-01    0.9616    0.969   0.9673
                                SNP      eQTL      sQTL    m6AQTL
estimated_group_prior     2.406e-04 8.895e-03  0.012934 1.236e-02
estimated_group_prior_var 1.858e+01 1.683e+01 36.589120 2.554e+01
estimated_group_pve       1.112e-01 8.236e-04  0.002867 7.999e-04
attributable_group_pve    9.612e-01 7.120e-03  0.024783 6.916e-03</code></pre>
<pre><code>$top1</code></pre>
<p><img src="figure/wbc_m6A_output.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Joining with `by = join_by(ensembl_gene_id)`</code></pre>
</div>
<div id="top-expressionsplicingm6a-units" class="section level3">
<h3>Top expression/splicing/m6A units</h3>
<p>For m6A or splicing QTLs, they are assigned to the nearest genes (m6A
needs to be confirmed with Kevin).</p>
<p><strong>Top 20 rows of SNPs or genes with PIP &gt; 0.6</strong></p>
<pre><code>$eQTL
     hgnc_symbol susie_pip type region_tag
1913     CSNK1G1    0.9587 eQTL      15_29
1916    RAPGEFL1    0.7362 eQTL      17_23
132       NDUFS2    0.6213 eQTL       1_81

$m6AQTL
     hgnc_symbol susie_pip   type region_tag
4938    SLC9A3R1    0.9539 m6AQTL      17_42
4922     ZKSCAN5    0.7863 m6AQTL       7_61
4067     THEMIS2    0.7816 m6AQTL       1_19
4312       TRAM2    0.7091 m6AQTL       6_39

$sQTL
     hgnc_symbol susie_pip type region_tag
4000    RN7SKP83    1.0000 sQTL       2_54
4011        CCM2    0.9932 sQTL       7_33
2385       ADPRH    0.7962 sQTL       3_74
4033       SLIT1    0.7387 sQTL      10_61
3788  ZNF225-AS1    0.6957 sQTL      19_30</code></pre>
</div>
<div id="top-m6a-modification-pip" class="section level3">
<h3>Top m6A modification pip</h3>
<pre><code>   hgnc_symbol region_tag susie_pip      z
1     SLC9A3R1      17_42    0.9539 -7.630
2      ZKSCAN5       7_61    0.7863  7.158
3      THEMIS2       1_19    0.7816  6.277
4        TRAM2       6_39    0.7091  5.233
5        BANF1      11_36    0.5786  6.174
6        TRIT1       1_25    0.5278  5.298
7        S1PR2       19_9    0.5220  9.939
8      WAC-AS1      10_20    0.4945 11.169
9       SQSTM1      5_108    0.4934 -4.857
10       CD320       19_8    0.3627 -4.062
11      SUPT5H      19_26    0.3551 -5.837
12       HMGN4       6_20    0.2707  4.297
13        MDM2      12_42    0.2706  4.606
14        SMG9      19_30    0.2544  3.901
15     CREB3L2       7_84    0.2449 -3.563
16       CNDP2      18_44    0.2407  3.360
17       CD79B      17_37    0.2406  5.091
18       CAND1      12_41    0.2377 -3.502
19      CHURC1      14_30    0.2315 -3.465
20       ADCY7      16_27    0.2181  3.717</code></pre>
<p><strong>Summing up PIPs for m6A peaks located in the same
gene</strong></p>
<p>Top 10 m6A PIPs by genes</p>
<pre><code># A tibble: 800 × 2
   hgnc_symbol total_susie_pip
   &lt;chr&gt;                 &lt;dbl&gt;
 1 SLC9A3R1              0.954
 2 ZKSCAN5               0.786
 3 THEMIS2               0.782
 4 TRAM2                 0.709
 5 BANF1                 0.579
 6 TRIT1                 0.528
 7 S1PR2                 0.522
 8 WAC-AS1               0.513
 9 SQSTM1                0.493
10 CD320                 0.379
# ℹ 790 more rows</code></pre>
</div>
<div id="top-splicing-pips" class="section level3">
<h3>Top splicing PIPs</h3>
<pre><code>                    peak_id hgnc_symbol region_tag susie_pip       z
1    chr2:85823772-85824227    RN7SKP83       2_54    1.0000   5.009
2    chr7:45009474-45009639        CCM2       7_33    0.9932 -11.719
3  chr3:119582452-119624602       ADPRH       3_74    0.7962   5.622
4   chr10:97007123-97023621       SLIT1      10_61    0.7387  -7.331
5   chr19:44112259-44118381  ZNF225-AS1      19_30    0.6957  -4.929
6  chr5:122111457-122130961      ZNF474       5_74    0.5628  -6.687
7   chr11:67120548-67124214       KDM2A      11_37    0.5416  -4.432
8    chr6:29693820-29694660       ZFP57       6_23    0.5000 -16.046
9    chr6:29694781-29695734   ZDHHC20P1       6_23    0.5000 -16.046
10  chr11:47761655-47765505       FNBP4      11_29    0.4878  10.101
11  chr19:13885521-13886291       BRME1      19_11    0.4797   6.500
12  chr19:13886427-13888866       BRME1      19_11    0.4797   6.500
13   chr7:72986365-72987174     STAG3L3       7_47    0.4741   6.870
14 chr1:224544695-224548197       CNIH3      1_116    0.4698   8.830
15    chr19:1036561-1037624        CNN2       19_2    0.4643   6.170
16    chr19:1036999-1037624        CNN2       19_2    0.4643  -6.170
17  chr17:47288203-47295101       ITGB3      17_28    0.4626  -4.041
18  chr19:49458856-49459455    ALDH16A1      19_34    0.4605  -4.118
19     chr7:5569315-5570155        ACTB        7_7    0.4288  -4.696
20  chr16:67690548-67690704       GFOD2      16_36    0.3912  -3.955</code></pre>
<p><strong>Summing up PIPs for spliced introns located in the same
gene</strong></p>
<p>Top 10 splicing PIPs by genes</p>
<pre><code># A tibble: 10 × 2
   hgnc_symbol total_susie_pip
   &lt;chr&gt;                 &lt;dbl&gt;
 1 RN7SKP83              1.00 
 2 CCM2                  0.993
 3 BRME1                 0.959
 4 CNN2                  0.929
 5 LINC02767             0.904
 6 LINC02834             0.811
 7 ADPRH                 0.796
 8 ZNF225-AS1            0.748
 9 SLIT1                 0.739
10 CNIH3                 0.690</code></pre>
</div>
<div id="top-genes-by-combined-pip" class="section level3">
<h3>Top genes by combined PIP</h3>
<pre><code>     hgnc_symbol combined_pip expression_pip splicing_pip  m6A_pip region_tag
2301    RN7SKP83        1.000        0.00000    1.0000000 0.000000       2_54
413         CCM2        0.993        0.00000    0.9932084 0.000000       7_33
610      CSNK1G1        0.969        0.95872    0.0000000 0.010146      15_29
285        BRME1        0.959        0.00000    0.4797074 0.000000      19_11
2601    SLC9A3R1        0.954        0.00000    0.0000000 0.953899      17_42
548         CNN2        0.929        0.00000    0.0001447 0.000000       19_2
1445   LINC02767        0.904        0.00000    0.2984697 0.000000      1_107
1448   LINC02834        0.811        0.00000    0.0247987 0.000000       9_41
65         ADPRH        0.796        0.00000    0.7961515 0.000000       3_74
3195     ZKSCAN5        0.786        0.00000    0.0000000 0.786335       7_61
2832     THEMIS2        0.782        0.00000    0.0000000 0.781582       1_19
2939       TRAM2        0.755        0.04565    0.0000000 0.709110       6_39
3214  ZNF225-AS1        0.748        0.00000    0.6956932 0.000000      19_30
2608       SLIT1        0.739        0.00000    0.7386585 0.000000      10_61
2226    RAPGEFL1        0.736        0.73623    0.0000000 0.000000      17_23
93      ALDH16A1        0.692        0.03207    0.1927717 0.006591      19_34
547        CNIH3        0.690        0.00000    0.4697789 0.000000      1_116
1772      NDUFS2        0.621        0.62125    0.0000000 0.000000       1_81
2869     TMEM156        0.606        0.17597    0.2148226 0.000000       4_32
411        CCDC9        0.597        0.59710    0.0000000 0.000000      19_33</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] cowplot_1.1.1        ggplot2_3.4.2        GenomicRanges_1.48.0
 [4] GenomeInfoDb_1.32.2  IRanges_2.30.1       S4Vectors_0.34.0    
 [7] BiocGenerics_0.42.0  ctwas_0.1.38         dplyr_1.1.2         
[10] workflowr_1.7.0     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.11            lattice_0.20-45        getPass_0.2-2         
 [4] ps_1.7.0               rprojroot_2.0.3        digest_0.6.33         
 [7] foreach_1.5.2          utf8_1.2.3             R6_2.5.1              
[10] evaluate_0.15          httr_1.4.6             highr_0.9             
[13] pillar_1.9.0           zlibbioc_1.42.0        rlang_1.1.1           
[16] rstudioapi_0.15.0      data.table_1.14.8      whisker_0.4           
[19] callr_3.7.3            jquerylib_0.1.4        Matrix_1.6-0          
[22] rmarkdown_2.14         labeling_0.4.2         stringr_1.5.0         
[25] RCurl_1.98-1.7         munsell_0.5.0          compiler_4.2.0        
[28] httpuv_1.6.5           xfun_0.30              pkgconfig_2.0.3       
[31] htmltools_0.5.2        tidyselect_1.2.0       tibble_3.2.1          
[34] GenomeInfoDbData_1.2.8 logging_0.10-108       codetools_0.2-18      
[37] fansi_1.0.4            withr_2.5.0            later_1.3.0           
[40] bitops_1.0-7           grid_4.2.0             jsonlite_1.8.7        
[43] gtable_0.3.3           lifecycle_1.0.3        git2r_0.30.1          
[46] magrittr_2.0.3         scales_1.2.1           cli_3.6.1             
[49] stringi_1.7.12         farver_2.1.1           XVector_0.36.0        
[52] fs_1.6.3               promises_1.2.0.1       pgenlibr_0.3.6        
[55] bslib_0.3.1            generics_0.1.3         vctrs_0.6.3           
[58] iterators_1.0.14       tools_4.2.0            glue_1.6.2            
[61] processx_3.8.0         fastmap_1.1.1          yaml_2.3.5            
[64] colorspace_2.1-0       knitr_1.39             sass_0.4.1            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
