---
title: "AD Enrichment Analysis"
output: html_document
date: '`r Sys.Date()`'
author: 'Jing Gu'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(data.table)
library(dplyr)
library(rtracklayer)
library(GenomicRanges)
source("~/scripts/R/utility_func.R")
source("~/projects/funcFinemapping/code/make_plots.R")
workdir <- "/project/xinhe/shared/GWAS_sumstat/AD"
```

## Estimating h2g enrichment with LDSC

GWAS: Bellenguez et al.(2022) harmonized summary statistics. 

LDSC parameters: 

* GWAS variants filtered by Hapmap3 SNPs
* LD computed based on Europeans from 1000 genome project
* Baseline annotations from BaselineLD_v2.2
* m6A annotations one at a time
```{r}
output <- fread("~/cluster/projects/ldsc_enrichment/AD/baselineLD_v2.2/m6A_hNPC.results")
baselines <- output$Category[grep("flanking", output$Category, invert = TRUE)]
DT::datatable(data.frame(annotations = gsub("L2_0", "", baselines)),
              caption="A list of baseline annotations:")
DT::datatable(
output %>% 
  filter(!grepl("flanking", Category)) %>%
  mutate(Category = gsub("L2_0", "", Category)) %>%
  select(Category, Enrichment, Enrichment_p, `Coefficient_z-score`) %>%
  arrange(Enrichment_p) %>%
  mutate(
    across(c(2,4), ~round(.x, 2)),
    Enrichment_p = format(Enrichment_p, scientific = TRUE, digits = 2)))
```

```{bash eval = FALSE, echo =FALSE}
# zcat 35379992-GCST90027158-MONDO_0004975.h.tsv.gz | cut -f 2,6,5,8,11,14,25,26 - > $workdir/Bellenguez2022/GCST90027158_AD_sumstats.txt

workdir=/project/xinhe/shared/GWAS_sumstat/AD
ldsc_base=/home/jinggu/cluster/data/gwas/gwas_ldsc/sldsc

python /home/jinggu/cluster/github/ldsc/munge_sumstats.py \
  --sumstats $workdir/schwartz/Kunkle_GWAS.txt.gz\
  --snp variant_id \
  --a1 effect_allele \
  --a2 other_allele \
  --frq effect_allele_frequency \
  --p GWAS_P \
  --N 64000 \
  --N-cas-col 22000 \
  --N-con-col 42000 \
  --signed-sumstats GWAS_BETA,0 \
  --chunksize 500000 \
  --merge-alleles $ldsc_base/1000G_EUR_QC.snplist \
  --out /home/jinggu/cluster/data/gwas/gwas_ldsc/AD_Kunkle
```

## Collecting m6A peaks
```{r}
f_peaks <- 
  list.files("~/cluster/data/features/raw/m6A_RELIC/",
             pattern="txt.gz")
for(i in f_peaks){
  f <- fread(paste0("~/cluster/data/features/raw/m6A_RELIC/", i))
  quantile(f$fdr)
  f <- 
    f %>% mutate(pos = gsub("\\[", ":", pos),
                 pos = gsub("\\]", "", pos),
                 pos = gsub("\\.", "\\*", pos))
  
  gr <- GRanges(f$pos)
  fname <- gsub(".txt.gz", ".bed", rev(strsplit(i, "=")[[1]])[1])
  write.table(data.frame(gr), 
              paste0("~/cluster/data/features/raw/m6A_RELIC/", fname),
              quote = F, sep = "\t",
              col.names = FALSE, row.names = FALSE)
}

dataDIR = "~/cluster/data/features/formatted/m6A"
fetal_m6a <- do.call(rbind, lapply(
  list.files(paste0(dataDIR, "/Xiao2019"), pattern="GSE114150"), function(i) {
  f <- fread(paste0(dataDIR, "/Xiao2019/", i))
  return(c(strsplit(i, "_")[[1]][2], quantile(f$V3-f$V2)))
}))
print("Fetal m6A peaks quantiles:")
data.frame(fetal_m6a)

adult_RELIC <- do.call(rbind, lapply(
  list.files(paste0(dataDIR, "/RELIC"), pattern=".bed")[-2:-1], function(i) {
  f <- fread(paste0(dataDIR, "/RELIC/", i))
  return(c(gsub(".bed", "", i), quantile(f$V3-f$V2)))
}))

adult_atlas <- do.call(rbind, lapply(
  list.files(paste0(dataDIR, "/m6A_atlas"), pattern=".bed")[-1], function(i) {
  f <- fread(paste0(dataDIR, "/m6A_atlas/", i))
  return(c(gsub(".bed", "", i), quantile(f$V3-f$V2)))
}))

print("Adult m6A peaks quantiles:")
DT::datatable(
  data.frame(rbind(adult_RELIC, adult_atlas))
)
```


## Enrichment results in barplots

* Labels on top of the bars: enrichment p-values 
* Legend for y-axis: annotation (percent of genome-wide variants that fall in the annotation peaks)

```{r}
trait_list <-
  c("LDL", "HDL", "bmi", "height", 
    "scz", "AD", "allergy", "coa_ukb")
annotations <- 
  read.table("/home/jinggu/cluster/projects/ldsc_enrichment/ldscore/annot_m6A_list.txt",
             header=FALSE)
# annot <- 
#   unlist(lapply(annotations$V1, function(i){
#           i = tolower(i)
#           i = gsub("_hg19.bed", "", i)
#           i = gsub("gse114150_", "m6A_fetal_", i)}))


annotations <- data.frame(annotations$V1,
                          gsub("m6A_", "", annotations$V1))
colnames(annotations)<-c("annot", "annotID")
annotations$annotID[c(12,13,15)] <- paste0("adult_", annotations$annotID[c(12,13,15)])
annotations <- 
  rbind(annotations, 
        data.frame(
         "annot" = gsub("_adult", "_atlas2", adult_atlas[,1]), 
        "annotID" = adult_atlas[,1])
  )
annotations_toplot <- annotations[-c(1,8, 14,15, 18),]
ldsc_tbl<-summarize_LDSC_outputs(traits = trait_list, 
                             referenceTab = annotations_toplot,
                             baseline="baselineLD_v2.2")
# order of annotations
annot_order<-annotations_toplot$annotID[
c(grep("adult", annotations_toplot$annotID),
  grep("fetal", annotations_toplot$annotID))]

lapply(seq(1, 8, 2), function(i){
  snp_enrichment_plot(ldsc_tbl %>% filter(trait %in% trait_list[i:(i+1)]), log.based = F, 
                    label.size = 3, tolabel = "pval",y.order = annot_order) + xlim(-40, 40) + facet_grid(. ~ trait) 
})

DT::datatable(ldsc_tbl %>% filter(trait %in% trait_list[5:6]) %>% 
                mutate(Prop_SNPs = format(scales::percent(Prop._SNPs)),
                       Prop_h2 = format(scales::percent(Prop._h2), digits=2),
                       Enrichment = format(Enrichment, digits=2),
                       Enrichment_p = format(Enrichment_p, digits=2)) %>%
                select(trait, Category, Prop_SNPs, Prop_h2, Enrichment, Enrichment_p))
```




